src/App.jsx

```diff
@@ -283,6 +304,17 @@ class App extends Component {
     this.setState({ showReadyLight: false });
   }
 
+  bypass2FA () {
+    const queryString = this.props.location.search;
+    const query = new URLSearchParams(queryString);
+    const cid = query.get('cid');
+    const voterDeviceId = VoterStore.voterDeviceId();
+    if (cid && cid !== voterDeviceId) {
+      VoterSessionActions.setVoterDeviceIdCookie(cid);
```

Comment:
I think in this case, it is fine to set the variable directly in the store. Presumably the bypass will happen before the detailed renders, and if it breaks something it will only be in a test script.